### PR TITLE
Fixes for 'Dedup - Generic v4' playbook

### PR DESCRIPTION
**Pull request by TalNos**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-8551

## Description
As part of the generic sub-playbook update effort, I reviewed 'Dedup - Generic v4' and noticed the following issues:
1- Automation inputs configurations - the script arguments on tasks number 1 & 27 were configured as 'simple'. changed all inputs to complex values.
2- Unclear task descriptions.
3- Unclear task name - task number 42 was named 'Find Similar Incidents By?', changed to 'Find Similar Incidents By Indicators or Fields?'
4- playbook description - the 'must' phrase appeared twice by mistake.
5- Missing playbook outputs - the 'DBotFindSimilarIncidentsByIndicators' context key was not listed as playbook output.

## screenshot
Dedup_-_Generic_v4

### CODE REVIEW

File: Packs/CommonPlaybooks/ReleaseNotes/2_4_7.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+#### Playbooks
+
+##### Dedup - Generic v4
+
+- Added the *'DBotFindSimilarIncidentsByIndicators'* context key to the playbook outputs.
+- Updated tasks and playbook descriptions.
+- Updated the script arguments on tasks number 1 and 27 from simple to complex values.
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/ReleaseNotes/2_4_7.md:8]: No need to mention the exact task's numbers

File: Packs/CommonPlaybooks/ReleaseNotes/2_4_7.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+- Updated the task name of task number 42.
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/ReleaseNotes/2_4_7.md:9]: No need for this - super minor change

Comment (TalNos) [reply Packs/CommonPlaybooks/ReleaseNotes/2_4_7.md]: removed

File: Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml [hunk 3]
```diff
@@ -733,8 +772,7 @@ tasks:
                 - - operator: greaterThanOrEqual
                     left:
                       value:
-                        simple: DBotFindSimilarIncidentsByIndicators.similarIncident.similarity
-                          indicators
+                        simple: DBotFindSimilarIncidentsByIndicators.similarIncident.similarity indicators
```
Comment (altmannyarden) [inline Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml:775]: There is a space between similarity and indicators.
As far as I know there are no spaces in context paths.

File: Packs/CommonPlaybooks/ReleaseNotes/2_4_12.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+#### Playbooks
+
+##### Dedup - Generic v4
+
+- Added the *'DBotFindSimilarIncidentsByIndicators'* context key to the playbook outputs.
+- Updated tasks and playbook descriptions.
+- Updated the script arguments of the playbook tasks from simple to complex values.
```
Comment (altmannyarden) [inline Packs/CommonPlaybooks/ReleaseNotes/2_4_12.md:8]: you can remove this line as its a yml change and not effecting the playbook itself.

Comment (TalNos) [reply Packs/CommonPlaybooks/ReleaseNotes/2_4_12.md]: removed

File: Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml (hunk 3 above)
Comment (TalNos) [reply Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml:775]: the context path includes a space between similarity and indicators.
Screenshot 2023-10-16 at 21 52 34

File: Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml [hunk 5]
```diff
@@ -56,23 +52,43 @@ tasks:
...
       minimunIncidentSimilarity:
-        simple: ${inputs.minimunIncidentSimilarity}
+        complex:
+          root: inputs.minimunIncidentSimilarity
       query:
-        simple: ${inputs.query}
+        complex:
+          root: inputs.query
       showIncidentSimilarityForAllFields:
         simple: ${inputs.showIncidentSimilarityForAllFields}
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml:88]: Let's make this one also complex no?

File: Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml [hunk 6]
```diff
@@ -166,15 +186,16 @@ tasks:
...
-    taskid: a58c04dc-bed0-47ed-8256-cfd360343866
+    taskid: b987c1f7-ae48-442b-835c-99b7d380cdb2
     type: regular
     task:
-      id: a58c04dc-bed0-47ed-8256-cfd360343866
+      id: b987c1f7-ae48-442b-835c-99b7d380cdb2
       version: -1
       name: Link similar incident
-      description: commands.local.cmd.linkIncidents
+      description: Link incidents together or mark them as duplicate and close them.
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml:198]: Was that changed by the formater or by you? if it's you, please leave it as it was.

File: Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml (hunk 5 above)
Comment (TalNos) [reply Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml:88]: done

File: Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml (hunk 6 above)
Comment (TalNos) [reply Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4.yml:198]: changed it back to the 'commands.local.cmd.linkIncidents' description.
Why this description was set for the task?

File: Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4_README.md [hunk 7]
```diff
@@ -1,50 +1,60 @@
...
+| query | The argument for dedicated query on incidents. This helps reduce the query size.<br/>Default \(same is in the Incident tab\): "-status:closed -category:job " | -status:closed -category:job | Optional |
 | closeReason | Specify the reason for closing the incident. This information will be added as a note/comment to the closed incident. | Closed by Dedup Playbook within inc ${incident.id} | Optional |
 
 ## Playbook Outputs
+
 ---
 
 | **Path** | **Description** | **Type** |
 | --- | --- | --- |
 | DBotFindSimilarIncidents | Return all the results from the "DBotFindSimilarIncidents" script. | string |
```
Comment (ssokolovich) [inline Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4_README.md:53]: This should be unknown right? returns a dict and not a string if I'm not mistaken.
For both outputs.

Comment (TalNos) [reply Packs/CommonPlaybooks/Playbooks/playbook-Dedup_-_Generic_v4_README.md:53]: changed the outputs type to 'unknown'

**Review by ssokolovich — APPROVED** [review]

**Pull request merged**